### DISCUSSION TITLE
AssertionError：vitis_hls not found in PATH

### DISCUSSION THREAD

**Discussion by DejianLin** [Xilinx/finn#1036] [Q&A] [answered]
Hi, I'm trying to run hardware generation step in "cnv_end2end_example" from bnn-pynq of finn repo and I am having this error;
![1712830709776](https://github.com/Xilinx/finn/assets/139863393/0cf4da56-b63d-44be-8ad5-021f1477099e)

I have already set the environment variables as follows:
![1712830074316](https://github.com/Xilinx/finn/assets/139863393/a438c2fa-d9fa-45c9-9f3b-ba395f609c23)

My version of Vivado is 2022，ubuntu 23.10 and I am a beginner in Linux.
Who can tell me what I should do？Thank you!!

**Comment by fpjentzsch** [comment, marked as answer]
Hi,

next to Vivado you need to install Vitis (HLS) as well.

**Reply by DejianLin** [reply]
Hi,

Thank you very much for your reply and I have installed Vitis_HLS:
![1712918787262](https://github.com/Xilinx/finn/assets/139863393/abe8490d-9651-4c94-9151-011af418fc4e)
environment variable：
![1712919115707](https://github.com/Xilinx/finn/assets/139863393/9262dc17-eb9a-4d90-82c9-36d7ac5d21f4)

I have also set relevant environment variables, but when I run tfc_end2end_example, the above error message still appears：
![1712919010464](https://github.com/Xilinx/finn/assets/139863393/a00afbe7-9c73-4771-9efe-abbaab025cd8)

I don't know the reason for the error and I hope you can give me some advice!!!

**Reply by AXT2185** [reply]
Have you solved the problem now? Cause I have the same problem and the same situation like you.  I have installed the Vitis(HLS) but still appear the problem.

**Reply by DejianLin** [reply]
Yes, I have already solved this problem. The reason is that there is an issue with my environment variable settings. I have been running finn with the root user, but my environment variables have no effect on the root user. Therefore, I also set environment variables for the root user.

**Reply by AXT2185** [reply]
Thanks your answer,  i solved the problem now.

**Reply by awaisk3** [reply]
How have you edited your environment variables. I'm facing same problem and I'm neither using finn/docker as root nor set environment variables as root. Then why am I facing these issues. Secondly how can I run finn as root user as I encounter many errors and warnings while I run it as root user

**Reply by DejianLin** [reply]
I executed the gedit ~/.bashrc command and set the environment variables like this:
![1716864868515](https://github.com/Xilinx/finn/assets/139863393/e4b24707-6601-4e39-90db-0c4792b189a8)
And every time I run finn, I will add the sudo command.
At last, I'm new to linux and don't know how to solve your problem.

**Reply by awaisk3** [reply]
Have you got 19 outputs ? I am stuck at number 12 and I need outputs to deploy on fpga.

Can you please contact me on my email: awaiss.kh18@gmail.com
Or WhatsApp: +92 336 6013018

I need little help,I've my open house project exhibition on Thursday and I'm stuck

**Reply by noisyor** [reply]
I am having the `AssertionError: vitis_hls not found in PATH` error. I am using Vivado and Vitis HLS 2023.2. 
@DejianLin did where did you add root in your environment variables? 
@awaisk3 did you solve the problem?
Also I am using Ubuntu 24.04.1 LTS inside WSL 2. The vivado software is loaded in C drive of windows.
My FINN_XILINX_PATH="/mnt/c/Xilinx" and FINN_XILINX_VERSION="2023.2"

**Reply by fpjentzsch** [reply]
See https://github.com/Xilinx/finn/discussions/1009#discussioncomment-13352446
